zerver/migrations/0282_userpresencenew_fill_data.py

```diff
@@ -0,0 +1,64 @@
+from django.db import migrations
+from django.db.backends.postgresql.schema import DatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+PRESENCE_ACTIVE_STATUS = 1
+PRESENCE_IDLE_STATUS = 2
+
+def fill_new_columns(apps: StateApps, schema_editor: DatabaseSchemaEditor) -> None:
+    UserPresence = apps.get_model('zerver', 'UserPresence')
+    UserPresenceNew = apps.get_model('zerver', 'UserPresenceNew')
+    # The lists below will have the latest (if any) presence of active/idle status for each user.
+    latest_active_presence_per_user = list(UserPresence.objects.filter(status=PRESENCE_ACTIVE_STATUS)
+                                           .order_by('user_profile', '-timestamp').distinct('user_profile'))
+    latest_idle_presence_per_user = list(UserPresence.objects.filter(status=PRESENCE_IDLE_STATUS)
+                                         .order_by('user_profile', '-timestamp').distinct('user_profile'))
+
+    user_profile_list = [
+        presence.user_profile for presence in latest_active_presence_per_user + latest_idle_presence_per_user
+    ]
+    user_id_to_user_profile = {user_profile.id: user_profile for user_profile in user_profile_list}
```

Comment:
I think this calculation will be very slow, since we're fetching related models.  All we need from `user_id_to_user_profile` is the `realm_id`, which is directly on `UserPresence`, so we should just be making a `user_id_to_realm_id` dictionary from the data we have without accessing `user_profile` at all.

Reply:
yup good catch, fixed now